### PR TITLE
ANDROID-11048 Several fixes in Sheet

### PR DESCRIPTION
**Pull request by gmerinojimenez**
### :goal_net: What's the goal?
Several fixes detected by design team

### ☑️ Checks
- [ ] I updated the documentation (readmes, wikis, etc). If no need to update documentation explain why.
- [ ] Tested with dark mode.
- [ ] Tested with API 21.

### :test_tube: How can I test this?
https://install.appcenter.ms/orgs/tuenti-organization/apps/android-core-7/distribution_groups/public
https://user-images.githubusercontent.com/4595241/190459532-9daf0585-bd93-4d0d-99f8-92710647f51f.mp4

### CODE REVIEW

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/CatalogMainActivity.kt [hunk 1]
```diff
@@ -95,7 +95,7 @@ class CatalogMainActivity : AppCompatActivity() {
             SectionItem("Tags", R.drawable.ic_tags, Section.TAG),
             SectionItem("Texts", R.drawable.ic_texts, Section.TEXTS),
             SectionItem("Titles", R.drawable.ic_texts, Section.TITLES),
-            SectionItem("Bottom Sheet", R.drawable.ic_bottomsheet, Section.BOTTOM_SHEET),
+            SectionItem("Sheet", R.drawable.ic_bottomsheet, Section.BOTTOM_SHEET),
```
Comment (gmerinojimenez) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/CatalogMainActivity.kt:98]: the official name is Sheet

File: library/build.gradle [hunk 2]
```diff
@@ -37,7 +37,7 @@ task sourceJar(type: Jar) {
 }
 
 dependencies {
-    api "com.google.android.material:material:1.4.0"
+    api "com.google.android.material:material:1.6.1"
```
Comment (gmerinojimenez) [inline library/build.gradle:40]: Needed to be able to setup the max height

File: catalog/src/main/res/layout/screen_fragment_bottom_sheet_catalog.xml [hunk 3]
```diff
@@ -56,6 +56,22 @@
...
+				android:layout_marginStart="8dp"
+				app:inputCheckText="Rows with icon" />
+
+		<com.telefonica.mistica.input.CheckBoxInput
+				android:id="@+id/input_bottomsheet_rows_with_description"
+				android:layout_width="match_parent"
+				android:layout_height="wrap_content"
+				android:layout_marginEnd="8dp"
+				android:layout_marginStart="8dp"
+				app:inputCheckText="Rows with description" />
```
Comment (gmerinojimenez) [inline catalog/src/main/res/layout/screen_fragment_bottom_sheet_catalog.xml:73]: new parameters to be able to test

File: library/src/main/res/layout/bottom_sheet_layout.xml [hunk 4]
```diff
@@ -14,56 +15,72 @@
...
+			android:textAppearance="@style/AppTheme.TextAppearance.Preset4"
+			/>
 
-		<TextView
-				android:id="@+id/title"
+	<androidx.core.widget.NestedScrollView
+			android:layout_width="match_parent"
+			android:layout_height="match_parent">
+		<LinearLayout
+				android:id="@+id/container_outer"
```
Comment (gmerinojimenez) [inline library/src/main/res/layout/bottom_sheet_layout.xml:37]: using nested scroll to be able to coordinate the scroll

**Review by yceballost — CHANGES_REQUESTED** [review]
1. Handle is not using `control` as a color constant. This cause a bad color in dark mode.
2. Is overlay using backgroundOverlay as a color constant?
3. Is it possible to make the animation smoother than now?
We found some curves for animation here
https://github.com/material-components/material-components-android/search?q=cubic+bezier 

nice job🙌

**Comment by gmerinojimenez** [comment]
@yceballost : https://github.com/Telefonica/mistica-android/pull/179/commits/01ee797f7e2fb0124df569cac6ecdb624d41bf0b

* Updated the handler color.
* Updated the icon in the catalog list

**Review by dpastor — APPROVED** [review]
Nice! 👍

**Review by dagonco — APPROVED** [review]
Niiice! 🚀

**Review by yceballost — APPROVED** [review]
nice job! ✨

**Pull request merged**